src/c1_state_machine/mod.rs

```diff
@@ -28,7 +28,7 @@ pub trait StateMachine {
 }
 
 /// A set of play users for experimenting with the multi-user state machines
-#[derive(Hash, Eq, PartialEq, Debug, Clone)]
+#[derive(Hash, Eq, PartialEq, Debug, Clone, Ord, PartialOrd)]
```

Comment:
I think we also won't need these traits if we use a HashMap for deduplicating

Reply:
true